GPSTest/src/main/java/com/chesapeaketechnology/gnssmonkey/service/GpsMonkeyService.java

```diff
@@ -188,58 +182,33 @@ public void start(InputSourceType inputSourceType) {
             this.inputSourceType = inputSourceType;
             Log.d(TAG, "GPSMonkey mode changed to " + inputSourceType.name());
             SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
-            prefs.edit().putInt(PREFS_CURRENT_INPUT_MODE, inputSourceType.ordinal()).commit();
+            prefs.edit().putInt(PREFS_CURRENT_INPUT_MODE, inputSourceType.ordinal()).apply();
```

Comment:
Are you sure we want to change this form commit to apply?  Actually, I am a bit surprised we are not seeing problems with using the default shared preferences from within the service thread.  Each thread has its own view of the Default Shared Preferences, which means that if one thread updates the preferences (e.g. the service thread), then other threads are not going to see that change until the app is closed, the preferences are written, and the app is reopened again.  This is also a problem because if multiple threads update the same preference there is not a resolution strategy to determine which thread has the correct value.

I even tried using the commit to force it to write the preferences to disk right away, but that did not force other threads to read from disk to get the latest preferences.

Reply:
I did some research on commit vs apply and the Android documentation actually says `SharedPreferences` instances are singletons within a process:
> As SharedPreferences instances are singletons within a process, it's safe to replace any instance of commit() with apply() if you were already ignoring the return value. ([source](https://developer.android.com/reference/android/content/SharedPreferences.Editor.html#apply()))
By default, all components of the same application run in the same process unless we specifiy a different process in the manifest using `android:process` ([source](https://developer.android.com/guide/components/processes-and-threads)). I checked out manifest and we don't have anything to change the process.
This [SO answer](https://stackoverflow.com/a/5960743) sums it up pretty well.